Some optimizations

alex-schneider opened the pull request.

Commit:
Faster `nextPow2()` function

Commit:
Type casts for function call

cespare commented:
Could you include benchstat results if the purpose of the changes is to make it faster? Thanks.

Commit:
Add benchmark

goos: linux
goarch: amd64
pkg: github.com/cespare/mph
cpu: Intel(R) Core(TM) i7-10510U CPU @ 1.80GHz

Old (3 runs w/o input `0` and `1`, because buggy):

Benchmark_nextPow2-8   	 7180598	       174.5 ns/op	       0 B/op	       0 allocs/op
Benchmark_nextPow2-8   	 6586764	       175.8 ns/op	       0 B/op	       0 allocs/op
Benchmark_nextPow2-8   	 6449841	       188.0 ns/op	       0 B/op	       0 allocs/op

New (3 runs w/o input `0` and `1`, no more buggy):

Benchmark_nextPow2-8   	11553169	       100.8 ns/op	       0 B/op	       0 allocs/op
Benchmark_nextPow2-8   	11764984	       105.3 ns/op	       0 B/op	       0 allocs/op
Benchmark_nextPow2-8   	10839817	        97.38 ns/op	       0 B/op	       0 allocs/op

New (3 runs w/ all input values):

Benchmark_nextPow2-8   	 9911341	       129.1 ns/op	       0 B/op	       0 allocs/op
Benchmark_nextPow2-8   	 8685609	       119.1 ns/op	       0 B/op	       0 allocs/op
Benchmark_nextPow2-8   	 8855067	       126.6 ns/op	       0 B/op	       0 allocs/op

cespare commented:
Thanks. But what I meant was that we should look at the effect on `Build`. We don't actually care how long `nextPow2` takes per se; we care how it affects `Build`, and `Build` only calls `nextPow2` twice (and then does a bunch of expensive stuff). So `nextPow2` would only matter if it were extremely slow.

Indeed, when I run the `Build` benchmark on my machine with and without this change, the difference isn't measurable in the noise:

```
name      old time/op  new time/op  delta
Build-24  34.5ms ± 4%  34.3ms ± 3%   ~     (p=0.067 n=34+31)
```
